// react-jsx-highcharts/dist/es/components/PlotLine/PlotLine.js
import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import _Object$getPrototypeOf from 'babel-runtime/core-js/object/get-prototype-of';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _createClass from 'babel-runtime/helpers/createClass';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React, { Component, Children, cloneElement, isValidElement } from 'react';
import PropTypes from 'prop-types';
import uuid from 'uuid/v4';
import isFunction from 'lodash-es/isFunction';
import attempt from 'lodash-es/attempt';
import Hidden from '../Hidden';
import getModifiedProps from '../../utils/getModifiedProps';

var PlotLine = function (_Component) {
  _inherits(PlotLine, _Component);

  function PlotLine(props) {
    _classCallCheck(this, PlotLine);

    var _this = _possibleConstructorReturn(this, (PlotLine.__proto__ || _Object$getPrototypeOf(PlotLine)).call(this, props));

    _this.getPlotLineConfig = function () {
      var _this$props = _this.props,
          id = _this$props.id,
          children = _this$props.children,
          rest = _objectWithoutProperties(_this$props, ['id', 'children']);

      if (!_this.id) {
        _this.id = isFunction(id) ? id() : id;
      }

      return _extends({
        id: _this.id
      }, rest);
    };

    _this.state = {
      rendered: false
    };
    return _this;
  }

  _createClass(PlotLine, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      var axis = this.props.getAxis();

      // Create Highcharts Plot Line on Axis
      var opts = this.getPlotLineConfig();
      axis.addPlotLine(opts);

      this.setState({
        rendered: true
      });
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate(prevProps) {
      if (getModifiedProps(prevProps, this.props) === false) return;

      var axis = this.props.getAxis();
      // Plot Lines cannot be updated, we have to remove and re-add
      var opts = this.getPlotLineConfig();
      axis.removePlotLine(opts.id);
      axis.addPlotLine(opts);
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      var axis = this.props.getAxis();
      attempt(axis.removePlotLine, this.id);
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var children = this.props.children;

      if (!children || !this.state.rendered) return null;

      var lineChildren = Children.map(children, function (child) {
        if (isValidElement(child) === false) return child;
        return cloneElement(child, { id: _this2.id });
      });

      return React.createElement(
        Hidden,
        null,
        lineChildren
      );
    }
  }]);

  return PlotLine;
}(Component);

PlotLine.defaultProps = {
  id: uuid
};


export default PlotLine;
