// react-jsx-highcharts/dist/es/components/Credits/Credits.js
import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import _Object$getPrototypeOf from 'babel-runtime/core-js/object/get-prototype-of';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _createClass from 'babel-runtime/helpers/createClass';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import { Component } from 'react';
import PropTypes from 'prop-types';
import attempt from 'lodash-es/attempt';
import getModifiedProps from '../../utils/getModifiedProps';

var Credits = function (_Component) {
  _inherits(Credits, _Component);

  function Credits() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, Credits);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = Credits.__proto__ || _Object$getPrototypeOf(Credits)).call.apply(_ref, [this].concat(args))), _this), _this.updateCredits = function (config) {
      var chart = _this.props.getChart();
      // Use default Highcharts value if text is not explicitly set
      if ('text' in config && !config.text) delete config.text;
      chart.addCredits(config, true);
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(Credits, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      var _props = this.props,
          text = _props.children,
          rest = _objectWithoutProperties(_props, ['children']);

      this.updateCredits(_extends({}, rest, {
        text: text
      }));
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate(prevProps) {
      var modifiedProps = getModifiedProps(prevProps, this.props, true);
      if (modifiedProps !== false) {
        this.updateCredits(modifiedProps);
      }
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      attempt(this.updateCredits, { enabled: false });
    }
  }, {
    key: 'render',
    value: function render() {
      return null;
    }
  }]);

  return Credits;
}(Component);

Credits.defaultProps = {
  enabled: true
};


export default Credits;
