// react-jsx-highcharts/dist/es/components/Legend/LegendTitle.js
import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import _Object$getPrototypeOf from 'babel-runtime/core-js/object/get-prototype-of';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _createClass from 'babel-runtime/helpers/createClass';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import { Component } from 'react';
import PropTypes from 'prop-types';
import attempt from 'lodash-es/attempt';
import getModifiedProps from '../../utils/getModifiedProps';

var LegendTitle = function (_Component) {
  _inherits(LegendTitle, _Component);

  function LegendTitle() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, LegendTitle);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = LegendTitle.__proto__ || _Object$getPrototypeOf(LegendTitle)).call.apply(_ref, [this].concat(args))), _this), _this.updateLegendTitle = function (config) {
      var chart = _this.props.getChart();
      chart.update({
        legend: {
          title: config
        }
      }, true);
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(LegendTitle, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      var _props = this.props,
          text = _props.children,
          rest = _objectWithoutProperties(_props, ['children']);

      this.updateLegendTitle(_extends({}, rest, {
        text: text
      }));
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate(prevProps) {
      var modifiedProps = getModifiedProps(prevProps, this.props, true);
      if (modifiedProps !== false) {
        this.updateLegendTitle(modifiedProps);
      }
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      attempt(this.updateLegendTitle, { text: null });
    }
  }, {
    key: 'render',
    value: function render() {
      return null;
    }
  }]);

  return LegendTitle;
}(Component);

export default LegendTitle;
